common/scripts/x86_64/Ubuntu_16.04/remote/unseal_vault.sh

```diff
@@ -1,33 +1,132 @@
 #! /bin/bash
 set -e
 
-export ADMIRAL_ENV="/etc/shippable/admiral.env"
-export BOOT_WAIT=10
-export VAULT_ADDR='http://127.0.0.1:8200'
+readonly VAULT_API_RESPONSE_FILE="/tmp/.vault_response"
+readonly ADMIRAL_ENV="/etc/shippable/admiral.env"
+readonly BOOT_WAIT=10
+
+##################### Begin Vault adapter ######################################
+
+__initialize() {
+  echo "Initializing vault api adapter"
+  RESPONSE_CODE=404
+  RESPONSE_DATA=""
+  CURL_EXIT_CODE=0
+
+  ## VAULT_URL is imported from this config
+  source $ADMIRAL_ENV
+
+  touch $VAULT_API_RESPONSE_FILE
+}
+
+__vault_get() {
+  __initialize
+
+  local url="$VAULT_URL/v1/$1"
+  {
+    RESPONSE_CODE=$(curl \
+      -X GET $url \
+      -H "Content-Type: application/json" \
+      --silent --write-out "%{http_code}\n" \
+      --output $VAULT_API_RESPONSE_FILE)
```

Comment:
Should this file be deleted before each request so we know for sure that the current contents are from the latest request?

Reply:
updated: the file is deleted in the initialize function